frontend/cs/r1cs/api.go

```diff
@@ -719,11 +716,27 @@ func (builder *builder) Commit(v ...frontend.Variable) (frontend.Variable, error
 			// it's the same variable ID, do nothing
 			continue
 		} else {
-			// append, it's a new variable ID
-			committed = append(committed, t.VID)
 			if t.VID < builder.cs.GetNbPublicVariables() {
 				nbPublicCommitted++
+			} else {
+				// Cannot commit to a secret variable that has already been committed to
```

Comment:
out of curiosity; why can't we just continue here and ignore that? Will comitting to a commitment help in anyway?

Reply:
Because of how the result of an `api.Commit` must be usable as a fiat-shamir challenge, it must have a "one-way" dependence on every variable input to this function. However, the way the commitment scheme is verified is Groth16 necessitates that every private variable can be used in one commitment at most. So if it were needed in multiple commitments, the next best thing is to commit to its commitment (commitments to commitments are easy, they're just conventional hashes.)